r2i/onnxrt_openvino/parameters.cc

```diff
@@ -0,0 +1,216 @@
+  };
+  parameter_map.emplace(std::make_pair(graph_optimization_level_meta.name,
+                                       graph_optimization_level_desc));
+
+  ParamDesc hardware_id_desc = {
+    {hardware_id_meta},
+    std::make_shared<HardwareIdAccessor>(this)
+  };
+  parameter_map.emplace(std::make_pair(hardware_id_meta.name, hardware_id_desc));
+}
+
+RuntimeError Parameters::Configure (std::shared_ptr<r2i::IEngine> in_engine,
+                                    std::shared_ptr<r2i::IModel> in_model) {
+  RuntimeError error;
+
+  if (nullptr == in_engine) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER, "Received null engine");
+    return error;
+  }
+
+  if (nullptr == in_model) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER, "Received null model");
+    return error;
+  }
+
+  auto engine = std::dynamic_pointer_cast<Engine, IEngine>(in_engine);
+  if (nullptr == engine) {
+    error.Set (RuntimeError::Code::INCOMPATIBLE_ENGINE,
+               "The provided engine is not an onnxrt engine");
+    return error;
+  }
+
+  auto model = std::dynamic_pointer_cast<onnxrt::Model, IModel>(in_model);
+  if (nullptr == model) {
+    error.Set (RuntimeError::Code::INCOMPATIBLE_MODEL,
+               "The provided engine is not an onnxrt model");
+    return error;
+  }
+
+  this->engine = engine;
+  this->model = model;
+
+  return error;
+}
+
+std::shared_ptr<r2i::IEngine> Parameters::GetEngine () {
+  return this->engine;
+}
+
+Parameters::ParamDesc Parameters::Validate (const std::string &in_parameter,
+    int type, const std::string &stype, RuntimeError &error) {
+  ParamDesc undefined = {{.name = "", .description = ""}, nullptr};
+
+  error.Clean ();
+
+  auto match = this->parameter_map.find (in_parameter);
+
+  /* The parameter wasn't found */
+  if (match == this->parameter_map.end ()) {
+    error.Set (RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+               "Parameter \"" + in_parameter + "\" does not exist");
+    return undefined;
+  }
+
+  ParamDesc param = match->second;
+
+  /* The parameter is not of the correct type */
+  if (param.meta.type != type) {
+    error.Set (RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+               "Parameter \"" + in_parameter + "\" is not of type " + stype);
+    return undefined;
+  }
+
+  return param;
+}
+
+RuntimeError Parameters::Get (const std::string &in_parameter, int &value) {
+  RuntimeError error;
+
+  ParamDesc param = this->Validate (in_parameter,
+                                    r2i::ParameterMeta::Type::INTEGER,
+                                    "integer", error);
+  if (error.IsError ()) {
+    return error;
+  }
+
+  auto accessor = std::dynamic_pointer_cast<IntAccessor>(param.accessor);
+
+  error = accessor->Get ();
+  if (error.IsError ()) {
+    return error;
+  }
+
+  value = accessor->value;
+  return error;
+}
+
+RuntimeError Parameters::Get (const std::string &in_parameter, double &value) {
+  RuntimeError error;
+  return error;
```

Comment:
Why isn't this implemented?

Shouldn't we report an error if someone tries to use this call?

Reply:
I agree, I'll return RuntimeError::Code::NOT_IMPLEMENTED in the Get() and Set() methods.